react/src/components/sections/planner/CourseInfoSubSection.js

```diff
@@ -0,0 +1,223 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { Link } from 'react-router-dom';
+import PropTypes from 'prop-types';
+import { withTranslation } from 'react-i18next';
+import axios from 'axios';
+
+import { appBoundClassNames as classNames } from '../../../common/boundClassNames';
+import qs from 'qs';
+import Scroller from '../../Scroller';
+import CloseButton from '../../CloseButton';
+import Divider from '../../Divider';
+
+import { clearCourseFocus, setLectures, setReviews } from '../../../actions/planner/courseFocus';
+import { addCourseRead } from '../../../actions/planner/list';
+
+import courseFocusShape from '../../../shapes/state/CourseFocusShape';
+import userShape from '../../../shapes/model/UserShape';
+
+import CourseSummarySubSection from './CourseSummarySubSection';
+import CourseReviewsSubSection from './CourseReviewSubSection';
+
+
+class CourseInfoSubSection extends Component {
+  constructor(props) {
+    super(props);
+
+    // eslint-disable-next-line fp/no-mutation
+    this.scoresRef = React.createRef();
+  }
+
+  componentDidMount() {
+    this._fetchLectures();
+    this._fetchReviews();
+  } //처음 코스 클릭시 과목 정보가 불러와지지 않는 오류를 위한 함수
+
+  componentDidUpdate(prevProps) {
+    const {
+      selectedListCode, courseFocus,
+      clearCourseFocusDispatch,
+    } = this.props;
+
+    if (prevProps.selectedListCode !== selectedListCode) {
+      clearCourseFocusDispatch();
+    }
+
+    if (!prevProps.courseFocus.course && courseFocus.course) {
+      this._fetchLectures();
+      this._fetchReviews();
+    }
+    if ((prevProps.courseFocus.course && courseFocus.course)
+    && (prevProps.courseFocus.course.id !== courseFocus.course.id)) {
+      this._fetchLectures();
+      this._fetchReviews();
+    }
+  }
+
+
+  _fetchLectures = () => {
+    const { courseFocus, setLecturesDispatch } = this.props;
+
+    axios.get(
+      `/api/courses/${courseFocus.course.id}/lectures`,
+      {
+        params: {
+          order: ['year', 'semester', 'class_no'],
+        },
+        metadata: {
+          gaCategory: 'Course',
+          gaVariable: 'GET Lectures / Instance',
+        },
+      },
+    )
+      .then((response) => {
+        const newProps = this.props;
+        if (newProps.courseFocus.course.id !== courseFocus.course.id) {
+          return;
+        }
+        setLecturesDispatch(response.data);
```

Comment:
다음처럼 개선할 수 있을 것 같습니다
```suggestion
        if (this.props.courseFocus.course.id === courseFocus.course.id)
          setLecturesDispatch(response.data);
```